app/controllers/api/storage.php

```diff
@@ -2,32 +2,338 @@
 
 use Appwrite\Auth\Auth;
 use Appwrite\ClamAV\Network;
-use Appwrite\OpenSSL\OpenSSL;
 use Appwrite\Utopia\Database\Validator\CustomId;
+use Appwrite\OpenSSL\OpenSSL;
 use Appwrite\Utopia\Response;
 use Utopia\App;
-use Utopia\Exception;
-use Utopia\Cache\Cache;
 use Utopia\Cache\Adapter\Filesystem;
+use Utopia\Cache\Cache;
 use Utopia\Config\Config;
-use Utopia\Database\Validator\Authorization;
 use Utopia\Database\Database;
 use Utopia\Database\Document;
+use Utopia\Database\Exception\Duplicate;
+use Utopia\Database\Exception\Duplicate as DuplicateException;
+use Utopia\Database\Exception\Structure as StructureException;
 use Utopia\Database\Query;
+use Utopia\Database\Validator\Authorization;
+use Utopia\Database\Validator\Permissions;
 use Utopia\Database\Validator\UID;
+use Appwrite\Extend\Exception;
 use Utopia\Image\Image;
+use Utopia\Storage\Compression\Algorithms\GZIP;
+use Utopia\Storage\Device\Local;
 use Utopia\Storage\Storage;
 use Utopia\Storage\Validator\File;
+use Utopia\Storage\Validator\FileExt;
 use Utopia\Storage\Validator\FileSize;
 use Utopia\Storage\Validator\Upload;
-use Utopia\Storage\Compression\Algorithms\GZIP;
 use Utopia\Validator\ArrayList;
-use Utopia\Validator\WhiteList;
+use Utopia\Validator\Boolean;
+use Utopia\Validator\HexColor;
+use Utopia\Validator\Integer;
 use Utopia\Validator\Range;
 use Utopia\Validator\Text;
-use Utopia\Validator\HexColor;
+use Utopia\Validator\WhiteList;
+
+App::post('/v1/storage/buckets')
+    ->desc('Create bucket')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.write')
+    ->label('event', 'storage.buckets.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'createBucket')
+    ->label('sdk.description', '/docs/references/storage/create-bucket.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUCKET)
+    ->param('bucketId', '', new CustomId(), 'Unique Id. Choose your own unique ID or pass the string `unique()` to auto generate it. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->param('name', '', new Text(128), 'Bucket name', false)
```

Comment:
```suggestion
    ->param('name', '', new Text(128), 'Bucket name')
```
the `optional` param is `false` by default. might be a good idea to skip it